private/bufpkg/bufmodule/bufmoduleref/bufmoduleref.go

```diff
@@ -69,6 +71,24 @@ type FileInfo interface {
 	isFileInfo()
 }
 
+// DigestChangedError is returned if module pin digests have changed unexpectedly.
+type DigestChangedError struct {
+	// CurrentDigest is the digest found in the buf.lock file.
+	CurrentDigest string
```

Comment:
Do we need to use these outside of this package/outside of the error message?

Reply:
Nope - I just used them in the tests but arguably that's testing internal type behavior. I'll unexport.